api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2499,8 +2499,33 @@ public void copyObject(CopyObjectArgs args)
             null,
             null,
             0)) {
-      // For now ignore the copyObjectResult, just read and parse it.
-      Xml.unmarshal(CopyObjectResult.class, response.body().charStream());
+
+      String bodyContent = new String(response.body().bytes(), StandardCharsets.UTF_8);
+      bodyContent = bodyContent.trim();
+      if (!bodyContent.isEmpty()) {
+        try {
+          CopyObjectResult result = Xml.unmarshal(CopyObjectResult.class, bodyContent);
+          return new ObjectWriteResponse(
+              response.headers(),
+              args.bucket(),
+              args.region(),
+              args.object(),
+              result.etag(),
+              response.header("x-amz-version-id"));
+        } catch (XmlParserException e) {
+          // As this CopyObjectResult REST call succeeded, just log it.
+          Logger.getLogger(MinioClient.class.getName())
+              .warning(
+                  "S3 service returned unknown XML for CopyObjectResult REST API. " + bodyContent);
+        }
+      }
+      return new ObjectWriteResponse(
+          response.headers(),
+          args.bucket(),
+          getRegion(args.bucket()),
```

Comment:
Do not call `getRegion()` for `ObjectWriteResponse`. You need to use `args.region()`. This is applicable everywhere.